libraries/AP_Landing/AP_Landing_Slope.cpp

```diff
@@ -289,19 +289,22 @@ void AP_Landing::type_slope_setup_landing_glide_slope(const Location &prev_WP_lo
         aim_height = flare_alt*2;
     }
 
-    // calculate time spent in flare assuming the sink rate reduces over time from sink_rate at aim_height
-    // to tecs_controller->get_land_sinkrate() at touchdown
-    const float weight = constrain_float(0.01f*(float)flare_effectivness_pct, 0.0f, 1.0f);
-    const float flare_sink_rate_avg = MAX(weight * tecs_Controller->get_land_sinkrate() + (1.0f - weight) * sink_rate, 0.1f);
-    const float flare_time = aim_height / flare_sink_rate_avg;
-
-    // distance to flare is based on ground speed, adjusted as we
-    // get closer. This takes into account the wind
-    float flare_distance = groundspeed * flare_time;
-
-    // don't allow the flare before half way along the final leg
-    if (flare_distance > total_distance*0.5f) {
-        flare_distance = total_distance*0.5f;
+    float flare_distance = 0;
+    if ((OptionsMask::LAND_POINT_IS_FLARE_POINT & _options) == 0) {
+        // calculate time spent in flare assuming the sink rate reduces over time from sink_rate at aim_height
+        // to tecs_controller->get_land_sinkrate() at touchdown
+        const float weight = constrain_float(0.01f*(float)flare_effectivness_pct, 0.0f, 1.0f);
+        const float flare_sink_rate_avg = MAX(weight * tecs_Controller->get_land_sinkrate() + (1.0f - weight) * sink_rate, 0.1f);
+        const float flare_time = aim_height / flare_sink_rate_avg;
+
+        // distance to flare is based on ground speed, adjusted as we
+        // get closer. This takes into account the wind
```

Comment:
This is a key comment; it doesn't matter how well you know the aircraft, the amount of distnace you spend flared is going to depend on the wind speed, which you will not take into account without this code.